sqlglot/dialects/tsql.py

```diff
@@ -324,6 +324,34 @@ def _parse(args: t.List[exp.Expression]) -> exp.Expression:
     return _parse
 
 
+# https://learn.microsoft.com/en-us/sql/t-sql/functions/parsename-transact-sql?view=sql-server-ver16
+def _build_parsename(args: t.List) -> t.Optional[exp.SplitPart]:
+    if len(args) != 2:
+        return None
+    arg_this: exp.Literal = seq_get(args, 0) or exp.Literal.string("")
+    arg_partnum: exp.Literal = seq_get(args, 1) or exp.Literal.number(1)
```

Comment:
On the other hand, I'm a bit skeptical about this. We don't do this argument counting pattern, because it doesn't facilitate partial parsing. We don't usually just `return None` in these builders. Perhaps it's better and more consistent to refactor this.